### PR TITLE
Compatibility with custom dataset

### PR DESCRIPTION
**Pull request by roskamsh**
This branch was created to make the targene-pipeline compatible with datasets that are not the UKBB.

### CODE REVIEW

**Comment by olivierlabayle** [comment]
Hello there! How are we doing around here? Is it ready for review? When ready you can add me as a reviewer on all your PRs :)

File: modules/genotypes.nf [hunk 1]
```diff
@@ -14,12 +14,23 @@ process filterBED{
 
     script:
         prefix = bedfiles[0].toString().minus('.bed')
-        """
-        TEMPD=\$(mktemp -d)
-        JULIA_DEPOT_PATH=\$TEMPD:/opt julia --project=/TargeneCore.jl --startup-file=no /TargeneCore.jl/bin/prepare_confounders.jl \
-        --input $prefix --output filtered.$prefix --qcfile $qcfile --maf-threshold $params.MAF_THRESHOLD --ld-blocks $ld_blocks --traits $traits filter
-        """
-
+
```
Comment (olivierlabayle) [inline modules/genotypes.nf:17]: Unless I am missing something, it is not necessary to duplicate the full script. The strategy that I have used so far is to declare a variable containing the option string or an empty string and include that variable in the script, see for instance: https://github.com/TARGENE/targene-pipeline/blob/a637cca5faf56fd0f9a539a492a57755ad2a447e/modules/tmle.nf#L36

**Review by olivierlabayle — COMMENTED** [review]
Thank you Breeshey for your work here. Appart from my comment on the script this looks good to me and the tests pass. Before merging we need to add one test for this new scenario and add some documentation. Happy to discuss that further :)

Todo:

- [ ] Add test
- [ ] Update docs

File: modules/genotypes.nf (hunk 1 above)
Comment (roskamsh) [reply modules/genotypes.nf]: This should now be updated to a cleaner version: [20d8bc9](https://github.com/TARGENE/targene-pipeline/commit/20d8bc9ef4d20790d5d9a8d91d03e2ce526b3fd9)

**Comment by roskamsh** [comment]
To-do checklist has now been completed.

File: test/non_ukbb.jl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# "local" profile assumes singularity is installed
```
[the remark this answers is not in the document]
Comment (roskamsh) [reply test/non_ukbb.jl]: This case has now been covered in the new test: efa437aa4b54b96eeb213896c7c94bcee877e1e8 and 8016bd056b4c977033b05523ea67c33411095cd3

File: main.nf [hunk 3]
```diff
@@ -95,14 +98,31 @@ workflow geneticConfounders {
...
 }
 
+workflow runPCA {
+    // Extract traits for UKBB
+    if (params.COHORT == "UKBB") {
+        extractTraits()
+        phenoInput = extractTraits.out
+    } else {
+        phenoInput = Channel.fromPath("$params.DECRYPTED_DATASET", checkIfExists: true)
+    }
```
[the remark this answers is not in the document]
Comment (roskamsh) [reply main.nf]: addressed here: 86bb82a3074cb24e50a39c6fda18f0e4e33c27c5

**Comment by olivierlabayle** [comment]
@roskamsh I've updated the new test configuration filename to match what it does, if the tests work, I'll merge and release! Thank you again for taking the time! :-)

**Review by olivierlabayle — APPROVED** [review]

**Pull request merged**